spec/dns_spec.rb

```diff
@@ -0,0 +1,118 @@
+describe LinuxAdmin::Dns do
+  RESOLV_CONF = <<DNS_END
+# Generated by NetworkManager
+search test.one.com test.two.com. test.com.
+nameserver 192.168.1.2
+nameserver 10.10.1.2
+nameserver 192.168.252.3
+DNS_END
+
+  SEARCH_ORDER = %w(test.one.com test.two.com. test.com.)
+  NAMESERVERS  = %w(192.168.1.2 10.10.1.2 192.168.252.3)
+
+  NEW_ORDER = %w(thing.com other.com)
+  NEW_NS    = %w(1.2.3.4 5.6.7.8)
+
+  FILE = "/etc/dns_file"
+
+  let(:dns) do
+    allow(File).to receive(:read).and_return(RESOLV_CONF)
+    described_class.new(FILE)
+  end
+
+  describe ".new" do
+    it "sets the filename" do
+      expect(dns.filename).to eq(FILE)
+    end
+
+    it "parses the nameservers" do
+      expect(dns.nameservers).to eq(NAMESERVERS)
+    end
+
+    it "parses the search order" do
+      expect(dns.search_order).to eq(SEARCH_ORDER)
+    end
+  end
+
+  describe "#reload" do
+    NEW_CONTENTS = <<DNS_END
+search thing.com other.com
+nameserver 1.2.3.4
+nameserver 5.6.7.8
+DNS_END
+
+    it "reloads the nameservers" do
+      expect(dns.nameservers).to eq(NAMESERVERS)
+
+      allow(File).to receive(:read).and_return(NEW_CONTENTS)
+      dns.reload
+
+      expect(dns.nameservers).to eq(NEW_NS)
+    end
+
+    it "reloads the search order" do
+      expect(dns.search_order).to eq(SEARCH_ORDER)
+
+      allow(File).to receive(:read).and_return(NEW_CONTENTS)
+      dns.reload
+
+      expect(dns.search_order).to eq(NEW_ORDER)
+    end
+  end
+
+  describe "#save" do
+    it "adds a search host" do
+      new_host = "new.server.com"
+      new_search_order = "search #{SEARCH_ORDER.join(' ')} #{new_host}"
+
+      expect(File).to receive(:write) do |file, contents|
+        expect(file).to eq(dns.filename)
+        expect(contents.include?(new_search_order)).to be true
+      end
+
+      dns.search_order.push(new_host)
+      dns.save
+    end
+
+    it "replaces the search order" do
+      old_search_order = "search #{SEARCH_ORDER.join(' ')}"
+      new_search_order = "search #{NEW_ORDER.join(' ')}"
+
+      expect(File).to receive(:write) do |file, contents|
+        expect(file).to eq(dns.filename)
+        expect(contents.include?(old_search_order)).to be false
+        expect(contents.include?(new_search_order)).to be true
```

Comment:
You don't need to do both the false and true here if you can compare them using eq()